src/wp-includes/media.php

```diff
@@ -5432,25 +5432,38 @@ function wp_get_webp_info( $filename ) {
  *                     that the `loading` attribute should be skipped.
  */
 function wp_get_loading_attr_default( $context ) {
-	// Only elements with 'the_content' or 'the_post_thumbnail' context have special handling.
-	if ( 'the_content' !== $context && 'the_post_thumbnail' !== $context ) {
-		return 'lazy';
+	// Skip lazy-loading for the overall block template, as it is handled more granularly.
+	if ( 'the_template' === $context ) {
+		return false;
 	}
 
-	// Only elements within the main query loop have special handling.
-	if ( is_admin() || ! in_the_loop() || ! is_main_query() ) {
-		return 'lazy';
+	// Do not lazy-load images in the header block template part, as they are likely above the fold.
+	$header_area = WP_TEMPLATE_PART_AREA_HEADER;
+	if ( "the_template_part_{$header_area}" === $context ) {
+		return false;
 	}
```

Comment:
Note for reviewers: These green lines are the only two things that are new in this function.

All remaining changes to this function are for code readability: I wrapped the existing code in the condition for `the_content` and `the_post_thumbnail`, and because of that GitHub makes it hard to decipher what's actually new here and what isn't. The logic for handling `the_content` and `the_post_thumbnail` is exactly the same as before.